regression-tests/sparktkregtests/testcases/frames/frame_quantile_test.py

```diff
@@ -0,0 +1,54 @@
+"""Tests the quantile functionality"""
+
+import unittest
+import sys
+import os
+from sparktkregtests.lib import sparktk_test
+from sparktk import dtypes
+
+
+class FrameQuantileTest(sparktk_test.SparkTKTestCase):
+    def setUp(self):
+        schema = [("value", int)]
+        self.frame_histogram = self.context.frame.import_csv(
+            self.get_file("histogram.csv"), schema=schema)
+
+    def test_histogram_standard(self):
+        """Tests the default behavior of histogram."""
+        result = self.frame_histogram.quantiles(
+            "value", [5, 10, 30, 70, 75, 80, 95])
+        values = result.take(result.row_count).data
+
+        # These values are known from the construction of the dataset
+        correct_values = [[5.0, 1.0], [10.0, 1.0], [30.0, 3.0], [70.0, 7.0],
+                          [75.0, 8.0], [80.0, 8.0], [95.0, 10.0]]
+        self.assertItemsEqual(values, correct_values)
+
+    @unittest.skip("Exception not raised by current impl")
+    def test_histogram_no_quantiles(self):
+        """Tests the behavior when no quantiles provided"""
+        with self.assertRaises(Exception):
+            self.frame_histogram.quantiles(
+                "value")
+
+    @unittest.skip("Exception not raised by current impl")
+    def test_histogram_exceed_quantile_range(self):
+        """Tests the behaviour when a quantile > 100 is requested"""
+        with self.assertRaises(Exception):
+            self.frame_histogram.quantiles(
+                "values", [101])
+
+    @unittest.skip("Exception not raised by current impl")
+    def test_histogram_negative_quantiles(self):
+        """Tests the behavior when negative quantiles provided"""
+        with self.assertRaises(Exception):
+            self.frame_histogram.quantiles(
+                "value", [-10, -5])
+
+    def test_histogram_bad_quantiles(self):
+        """Tests the behavior when string quantiles provided"""
+        with self.assertRaises(Exception):
```

Comment:
I cannot use assertRaisesRegexp in these because the function is buggy and does not raise an exception right now.
I am not sure what type of message will be displayed.

From: Lewis Coates [mailto:notifications@github.com]
Sent: Thursday, August 11, 2016 4:16 PM
To: trustedanalytics/spark-tk <spark-tk@noreply.github.com>
Cc: Dwivedi, Aayushi <aayushi.dwivedi@intel.com>; Author <author@noreply.github.com>
Subject: Re: [trustedanalytics/spark-tk] Merge Stats quantile join (#100)


In regression-tests/sparktkregtests/testcases/frames/frame_quantile_test.py<https://github.com/trustedanalytics/spark-tk/pull/100#discussion_r74521000>:

> +    def test_histogram_exceed_quantile_range(self):

> +        """Tests the behaviour when a quantile > 100 is requested"""

> +        with self.assertRaises(Exception):

> +            self.frame_histogram.quantiles(

> +                "values", [101])

> +

> +    @unittest.skip("Exception not raised by current impl")

> +    def test_histogram_negative_quantiles(self):

> +        """Tests the behavior when negative quantiles provided"""

> +        with self.assertRaises(Exception):

> +            self.frame_histogram.quantiles(

> +                "value", [-10, -5])

> +

> +    def test_histogram_bad_quantiles(self):

> +        """Tests the behavior when string quantiles provided"""

> +        with self.assertRaises(Exception):

RaisesRegexp

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/trustedanalytics/spark-tk/pull/100/files/bb2cac0e184db3e639350db978f3a2ae058bfd24#r74521000>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AMHWjkuBlYCQc_sOc8WTMJCamnPLiOwvks5qe602gaJpZM4JilJ_>.